spec/steps/claim_details/form_spec.rb

```diff
@@ -105,4 +120,60 @@
       end
     end
   end
+
+  context 'when work_before is yes' do
```

Comment:
so this is still missing a number of edge cases (specifically it has no tests for an y of the invalid criteria. Lets spend some time this afternoon paring on this...